pkg/api/types.go

```diff
@@ -1380,3 +1381,15 @@ const (
 
 	PortHeader = "port"
 )
+
+// ResourceStats represents compute resource usage metrics associated with a pod or a container.
+type ResourceStats struct {
+	// Detailed instantaneous metrics.
+	Detailed *cadvisor_api.ContainerInfo `json:"detailed,omitempty"`
```

Comment:
IIUC, the only reason you touched pkg/api/types.go here was because Kubelet is reusing the types from master. Should we introduce a new API types file specifically for the kubelet, like @abhgupta did in #4674 for scheduler? Or maybe we can have three types files, one for master-only, one for kubelet-only, and one for shared (similar to how we factored the proto bufs for our internal system)? I think it's fine to share some data structures between master and kubelet as long as we think carefully about upgrades/evolution. The two are going to be talking to each other anyway so there's no escaping thinking about backward compatibility no matter what you do.